### DISCUSSION TITLE
Postgres like search with strings

### DISCUSSION THREAD

**Discussion by kiranbadi** [hasura/graphql-engine#9455] [Question]
Is it possible to do something like below with Hasura

`query SearchUsers($name: String!) {
  users(where: { name: { _like: "%" + $name + "%" } }) {
    id
    name
    email
  }
}`

I am getting  Cannot parse unexpected character "+" . Is this something possible ?

**Comment by BenoitRanque** [comment]
The GraphQL spec does not have support for in-line string concatenation, which is what you're showing here.

But what you can do instead is concat the strings yourself before passing them to your client of choice.

In javascript this could look like this:


```js
const nameToSearch = '' // string from user input

const query = `query SearchUsers($name: String!) { users(where: { name: { _like: $name } }) { id name email } }`
const variables = { name: `%${nameToSearch}%` }

const response = await executeGraphQL(query, variables)
```

**Reply by kiranbadi** [reply]
Thank you Benoit for your reply.

We use Hasura graphql console for building the rest endpoints and then we provide those endpoints to UI team and definitely we can do it via the way you suggested. In fact I was thinking in that direction as well as building the postgres function if nothing works. Thank you.

**Reply by BenoitRanque** [reply]
Yes, you can abstract away the postgres-specific wildcard [using a postgres function](https://hasura.io/docs/latest/schema/postgres/custom-functions/#example-text-search-functions).

Please note, using wildcards and the LIKE operator is not a substitute for [full text search](https://www.postgresql.org/docs/current/textsearch.html), and [may perform poorly](https://use-the-index-luke.com/sql/where-clause/searching-for-ranges/like-performance-tuning).

**Comment by kiranbadi** [comment]
I am aware of this Benoit.
